### PR TITLE
refactor(router): Remove internal state tracking for browserUrlTree

### PR DESCRIPTION
**Pull request by atscott**
The `browserUrlTree` is _only_ used to support the `onSameUrlNavigation: 'ignore'` logic. We can achieve this functionality without having this state tracked inside the Router. Instead, we can re-examine what `ignore` means: We don't want to rerun the matching logic, guards, or resolvers when we already know that nothing is changing.
Outside of the "navigated", there are two things that constitute a "change":

1. The browser URL might change. Because of `skipLocationChange`, the browser URL might not always match the internal state of the Router (we can navigate to a path but skip updating the browser URL). If we're navigating to a place that would change the browser URL, we should process the navigation. Theoretically, all we need to really do is update the browser URL instead of processing the whole navigation w/ guards, redirects, and resolvers. But this doesn't matter that much because the default value for `runGuardsAndResolvers` will skip all of this anyways.
2. The internal state of the Router might change. That is, we're navigating to a new path and may or may not be updating the updating the browser URL.

If either of the above are true, we process the navigation. If both are false, we aren't changing anything so we can safely ignore the navigation request (as long as `onSameUrlNavigation === 'ignore'`).

Why is this change important?

* Simplification of Router internals. The Router has a lot of special case handling and one-offs to handle a limited set of scenarios. Removing these when possible makes the code easier to follow

### CODE REVIEW

**Comment by atscott** [comment]
[Green TGP](https://fusion2.corp.google.com/presubmit/562058730/OCL:562058730:BASE:562201086:1693666791927:7409d098;groups=PossiblyNewlyFailing,Passing/targets)

File: packages/router/src/navigation_transition.ts [hunk 1]
```diff
@@ -393,15 +393,28 @@ export class NavigationTransitions {
...
+                               t.extractedUrl.toString() !== t.currentUrlTree.toString();
+
+                           // We're updating the browser URL to something new (navigation is going
+                           // to somewhere not displayed in the URL bar and we will update the URL
+                           // bar if navigation succeeds)
+                           const extractedBrowserUrl = this.urlHandlingStrategy.extract(
+                               this.urlSerializer.parse(this.location.path(true)));
+                           const updatingBrowserUrl =
+                               extractedBrowserUrl.toString() !== t.extractedUrl.toString() &&
+                               !t.extras.skipLocationChange;
```
Comment (AndrewKushnir) [inline packages/router/src/navigation_transition.ts:414]: nit: those operations are probably not expensive, but we'll be doing them always eagerly now (this was done more lazily before). As an option, we can move them to separate functions.

**Review by AndrewKushnir — APPROVED** [review]
@atscott looks great 👍 Just left one comment, but it's **not** a blocking one, feel free to ignore.

File: packages/router/src/navigation_transition.ts (hunk 1 above)
Comment (atscott) [reply packages/router/src/navigation_transition.ts:414]: 👍 Done

**Review by AndrewKushnir — APPROVED** [review]
👍

**Comment by atscott** [comment]
caretaker note: Please merge and sync on its own and wait until not a Friday.

**Pull request closed without merging**